### PR TITLE
Fix/install GitHub submodule warn

### PR DESCRIPTION
**Pull request by ashander**
Add warning when downloading from github repo that uses submodules 
(assessed through presence of .gitsubmodules file). 

Help for install_github updated to suggest use of install_git with proper args in this case.

Limited testing (with mock responses from download and attempt to download .gitsubmodules).

Partially responsive to #737

### CODE REVIEW

**Comment by ashander** [comment]
Travis CI fails with message below. Seems like a separate issue.

````sh
Warning message:

package ‘BiocInstaller’ is not available (for R version 3.1.3)

The command "Rscript -e 'options(repos = "http://cran.rstudio.com"); tryCatch({ deps <- devtools::install_deps(dependencies = TRUE) }, error = function(e) { message(e); q(status=1) }); if (!all(deps %in% installed.packages())) { q(status = 1, save = "no") }'" failed and exited with 1 during .
````

**Comment by ashander** [comment]
Yes looks like @jimhester has a fix in PR #746

File: R/install-github.r [hunk 1]
```diff
@@ -98,6 +106,10 @@ remote_download.github_remote <- function(x, quiet = FALSE) {
   } else {
     auth <- NULL
   }
+  ## if repo uses submodules warn user
```
Comment (hadley) [inline R/install-github.r]: 1. I think the comment is extraneous (given how simple the code is)
2. Can you please use a space after `if`
3. `warning()` looks like it's indented too much
4. I always use `call. = FALSE` with warning

Comment (ashander) [inline R/install-github.r]: Thanks! Sorry for missing the indent and if-paren style with this PR. 





I've made the changes, and also fixed indent style in changes the tests






- Jaime

**Comment by hadley** [comment]
Looks good! Could you please add a bullet point to NEWS, thanking your github username?

(FYI github only notifies me about comments, not commits, so when done please add a comment like "PTAL")

**Comment by ashander** [comment]
Done! PTAL



- Jaime

**Comment by hadley** [comment]
Thanks!

**Comment by jonkeane** [comment]
I'm attempting to install from a github repository that has a submodule. I've tried to use `install_git('[url to repo]", args="--recursive")` but I get a warning that args is deprecated. Is there a new prefered method for installing submodules? Now that `args` is deprecated with `install_git()`?

Sorry if reviving this issue is inappropriate, I'm happy to start a new one if that is preferred.

**Comment by ashander** [comment]
I can't say for certain. But, it seems adopting `git2r` traded stability for the args. See https://github.com/hadley/devtools/commit/1b7e67acce1398f2a409752f7d99c9a0220196a1

Rather than commenting here, it's probably better to open a new issue regarding git2r and args (unless that discussion has already happened -- I haven't searched). In that issue you could reference this PR.

In any case, the suggestion to use `install_git` with `--args` should be removed from the docs to `install-github.r`

**Comment by jimhester** [comment]
@jonkeane `install_git()` uses libgit2 rather than the git client to clone the git repository. [git2r](https://github.com/ropensci/git2r) the R client to the libgit2 library does not look it has an interface for submodules on cursory inspection.

Probably the easiest thing to do assuming you do have the command line git client installed is define a simple helper function to clone the repository locally and install from there.

```r
install_submodule_git <- function(x, ...) {
  install_dir <- tempfile()
  system(paste("git clone --recursive", shQuote(x), shQuote(install_dir)))
  devtools::install(install_dir, ...)
}
install_submodule_git("https://github.com/jonkeane/mocapGrip")
list.files(system.file(package = "mocapGrip", "python", "pyelan"))
#> [1] "__init__.py"      "elanGen.py"       "elanSkeleton.eaf" "pyelan.py"        "README.md"        "relPathFix.py"

**Pull request merged**